gorm-tools/src/main/groovy/gorm/tools/problem/ProblemHandler.groovy

```diff
@@ -99,7 +101,12 @@ class ProblemHandler {
                 log.error("MAYBE UNEXPECTED? Data Access Exception ${msgInfo}", StackTraceUtils.deepSanitize(e))
                 return DataProblem.of(e)
             }
-        } else {
+        } else if (e instanceof HttpMessageNotReadableException || e instanceof JsonException) {
+            //this happens if request contains bad data / malformed json we dont want to log stacktraces for these as they are expected
+            return GenericProblem.of(e)
```

Comment:
do we do this elsewhere? 
everything is a DataProblem, Doesnt Problem.of do the same thing here?
Shouldn't this be a DataProblem anyway?
Where are we doing integration testing for the common occurence of this?  Does not need to be in domain9, we have int and REST tests in rally-api or testify.